lib/utils/fields.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils"
+)
+
+// Fields represents a generic string-keyed map.
+type Fields map[string]interface{}
+
+// GetString returns a string representation of a field.
+func (f Fields) GetString(key string) string {
+	val, found := f[key]
+	if !found {
+		return ""
+	}
+	v, _ := val.(string)
```

Comment:
Why are we ignoring `ok` here but not in the other `GetX` methods? If we really don't need to check it here maybe just use `v := val.(string)`

Reply:
Changed to match your suggestion.